File: src/prototype/Fortran/cost.f90
!! Calcul le cout d'une solution m, f sur al grille centrée

subroutine cost(R,m,f,N,Q)
    implicit none
    integer, intent(in) :: N,Q;
    double precision, dimension(N+1,Q+1), intent(in) :: m,f; 
    double precision, intent(out) :: R;

    R = sum(m**2/(1.0*f));
end subroutine

File: src/prototype/test2D/cost.m
function R = cost(V)
    globals;
    R = (V(1:(N+1)*(P+1)*(Q+1)).^2 + V((N+1)*(P+1)*(Q+1)+1:2*(N+1)*(P+1)*(Q+1)).^2);
    R = R./V(2*(N+1)*(P+1)*(Q+1):end);
    R = sum(R);
end

